HotelManagement/src/core/Room.java

```diff
@@ -34,25 +47,198 @@ private void setNumber(int number) {
 		this.number = number;
 	}
 
+	public List<Booking> getBookings() {
+		return bookings;
+	}
+
 	/**
 	 * Returns {@code true} if the room is booked.
 	 * @return {@code true} if the room is booked.
 	 */
-	public boolean isBooked() {
-		return isBooked;
+	public boolean isBooked(LocalDate fromDate, LocalDate toDate, int size) {
+		ArrayList<String> availableDates = findAvailableDatesForIntervalAndSize(fromDate, toDate, size);
+		String date = fromDate.getYear() + "-" + fromDate.getMonthValue() + "-" + fromDate.getDayOfMonth() + " to " +
+			toDate.getYear() + "-" + toDate.getMonthValue() + "-" + toDate.getDayOfMonth();
+		if (availableDates == null) {
+			return false;
+		}
+
+		return true;
 	}
 
-	/**
-	 * Books the room.
-	 */
-	public void bookTheRoom() {
-		this.isBooked = true;
+	public Set<AbstractCommodity> getCommodities() {
+		return commodities;
 	}
 
-	/**
-	 * Clear booking of the room.
-	 */
-	public void clearRoomBooking() {
-		this.isBooked = false;
+	public void prepareRoomCommodities() {
+		for(AbstractCommodity commodity : commodities) {
+			commodity.prepare();
+		}
+
+		maintenanceDates.add(LocalDateTime.now());
+	}
+
+	public int getNumberOfBeds() {
+		int numberOfBeds = 0;
+		for (AbstractCommodity commodity : commodities) {
+			if (commodity instanceof Bed) {
+				numberOfBeds++;
+			}
+		}
+
+		return numberOfBeds;
+	}
+
+	public void createBooking(LocalDate fromDate, LocalDate toDate, int size, String guestName, String guestId) {
+		ArrayList<String> availableDates = findAvailableDatesForIntervalAndSize(fromDate, toDate, size);
+		String date = fromDate.getYear() + "-" + fromDate.getMonthValue() + "-" + fromDate.getDayOfMonth() + " to " +
```

Comment:
See if you can parse this in a better way
this amounts to 2020-12-1-2021-12-1. Separate the "from" dates from the "to" dates. Or even better find how you can use LocalDate's methods for doing this. Will be way easier and cleaner.